module-system/module-implementations/sov-evm/src/evm/error/rpc.rs

```diff
@@ -0,0 +1,642 @@
+//! Implementation specific Errors for the `eth_` namespace.
+
+use std::time::Duration;
+
+use jsonrpsee::core::Error as RpcError;
+use jsonrpsee::types::error::CALL_EXECUTION_FAILED_CODE;
+use jsonrpsee::types::ErrorObject;
+use reth_interfaces::RethError;
+use reth_primitives::abi::decode_revert_reason;
+use reth_primitives::{Address, Bytes, U256};
+use reth_revm::tracing::js::JsInspectorError;
+use reth_rpc_types::error::EthRpcErrorCode;
+use reth_rpc_types::{BlockError, CallInputError};
+use revm::primitives::{EVMError, ExecutionResult, Halt, InvalidHeader, OutOfGasError};
+
+use super::pool::{
+    Eip4844PoolTransactionError, InvalidPoolTransactionError, PoolError, PoolTransactionError,
+};
+use super::result::{internal_rpc_err, invalid_params_rpc_err, rpc_err, rpc_error_with_code};
+
+/// Result alias
+pub type EthResult<T> = Result<T, EthApiError>;
+
+/// Errors that can occur when interacting with the `eth_` namespace
+#[derive(Debug, thiserror::Error)]
+#[allow(missing_docs)]
+pub enum EthApiError {
+    /// When a raw transaction is empty
+    #[error("Empty transaction data")]
+    EmptyRawTransactionData,
+    #[error("Failed to decode signed transaction")]
+    FailedToDecodeSignedTransaction,
```

Comment:
We have `EmptyRawTransactionData, FailedToDecodeSignedTransaction,  FailedToRecoverSigner,` in `evm/primitive_types/RawEvmTxConversionError`. Let's replace `RawEvmTxConversionError` with `EthApiError`.